### PR TITLE
Feature/1421 dont store empty answers

### PR DESCRIPTION
**Pull request by ifirmawan**
## TODO/DONE

- [x] empty payload test in device sync
- [x] empty payload test in webform pending data submission

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207178470640775

### CODE REVIEW

File: backend/api/v1/v1_mobile/tests/tests_api_sync_empty_payload.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    def test_empty_non_required_autofield_payload(self):
+        mobile_adm = self.mobile_user.administrations.first()
+        payload = {
+            "formId": self.form.id,
+            "name": "datapoint #1",
+            "duration": 1,
+            "submittedAt": "2024-04-29T02:38:13.807Z",
+            "submitter": self.mobile_user.name,
+            "geo": [6.2088, 106.8456],
```
Comment (dedenbangkit) [inline backend/api/v1/v1_mobile/tests/tests_api_sync_empty_payload.py:76]: what happened if we send `[]` @ifirmawan? I see in the mobile test that we are able to send an empty array for geo

File: backend/api/v1/v1_mobile/tests/tests_api_sync_empty_payload.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                102: ["male"],
+                103: 62723817,
+                104: mobile_adm.id,
+                105: [6.2088, 106.8456],
+                106: ["wife__husband__partner"],
+                107: "photo.jpeg",
+                108: "2024-04-29",
+                109: 0.6,
+                110: "72b9ecb2-c400-4b76-bcba-0a70a6942bb6",
+                111: ""
```
Comment (dedenbangkit) [inline backend/api/v1/v1_mobile/tests/tests_api_sync_empty_payload.py:89]: what if required but not visible (e.g [grade claim](https://github.com/akvo/rtmis/blob/7b8296deec684dc549551bbd08b75bfba69a40eb/backend/source/forms/1699354006503.prod.json#L1777))?

File: backend/api/v1/v1_mobile/tests/tests_api_sync_empty_payload.py (hunk 1 above)
Comment (ifirmawan) [reply backend/api/v1/v1_mobile/tests/tests_api_sync_empty_payload.py:145]: Regarding the empty geo,  I just found out that it's coming from the data payload, not from the answer
```
{"answers": {"1701757876670": 50664, "1701757914033": ["new"], "1706105071390": [32.2989983, -25.68916], "1713505867": 13}, "duration": 1, "formId": 1701757876668, "geo": [], "name": "Moruamekwi", "submission_type": 1, "submittedAt": "2024-04-29T05:59:54.010Z", "submitter": "test empty"}
```
I've fixed it on this commit: 51c93b12f1284b7048882ff080f6a5436647def9

File: backend/api/v1/v1_mobile/tests/tests_api_sync_empty_payload.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            payload,
+            follow=True,
+            content_type="application/json",
+            **{"HTTP_AUTHORIZATION": f"Bearer {self.token}"},
+        )
+        data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn("101", data["message"])
+        self.assertIn("value may not be null.", data["message"])
+        self.assertIn("106", data["message"])
```
Comment (dedenbangkit) [inline backend/api/v1/v1_mobile/tests/tests_api_sync_empty_payload.py:66]: self.assertFalse(Answers.objects.filter(question=101,uuid="72b9ecb2-c400-4b76-bcba-0a70a6942bb6"))

**Review by dedenbangkit — CHANGES_REQUESTED** [review]

**Review by dedenbangkit — APPROVED** [review]

**Pull request merged**